pkg/queue/request_metric.go

```diff
@@ -126,17 +130,21 @@ func (h *requestMetricsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request
 			panic(err)
 		}
 		ctx := metrics.AugmentWithResponseAndRouteTag(h.statsCtx,
-			rr.ResponseCode, routeTag)
+			dw.ResponseCode, routeTag)
 		pkgmetrics.RecordBatch(ctx, requestCountM.M(1),
 			responseTimeInMsecM.M(float64(latency.Milliseconds())))
+		if h.reportLatencyInHeader {
+			dw.Header().Add("queue_proxy_request_latency_ms", strconv.Itoa(int(latency.Milliseconds())))
+		}
+		dw.RealFlush()
```

Comment:
`writer.WriteHeader` is called inside `dw.RealFlush`, I have to do this because the header was modified at line 137, and `writer.WriteHeader` has to be called after the header modification for the newly added header to take effect.